src/ToolBox/SOS/Strike/strike.cpp

```diff
@@ -331,14 +331,22 @@ DEBUG_STACK_FRAME g_Frames[MAX_STACK_FRAMES];
 AMD64_CONTEXT g_X64FrameContexts[MAX_STACK_FRAMES];
 
 static HRESULT
-GetContextStackTrace(PULONG pnumFrames)
+GetContextStackTrace(ULONG osThreadId, PULONG pnumFrames)
 {
     PDEBUG_CONTROL4 debugControl4;
     HRESULT hr;
 
     // Do we have advanced capability?
     if ((hr = g_ExtControl->QueryInterface(__uuidof(IDebugControl4), (void **)&debugControl4)) == S_OK)
     {
+        ULONG oldId, id;
+        g_ExtSystem->GetCurrentThreadId(&oldId);
```

Comment:
Can you please explain why we need do to this dance with swapping the thread id?